GOT-StudentVersion/GOT-StudentVersion/ViewController.swift

```diff
@@ -8,18 +8,102 @@
 
 import UIKit
 
-class ViewController: UIViewController {
+class ViewController: UIViewController{
+    
+    @IBOutlet weak var tableView: UITableView!
+    
+    
+    //    var seasonOne = GOTEpisode.allEpisodes.filter {$0.season == 1}
+    //    var seasonTwo = GOTEpisode.allEpisodes.filter {$0.season == 2}
+    //    var seasonThree = GOTEpisode.allEpisodes.filter {$0.season == 3}
+    //    var seasonFour = GOTEpisode.allEpisodes.filter {$0.season == 4}
+    //    var seasonFive = GOTEpisode.allEpisodes.filter {$0.season == 5}
+    //    var seasonSix = GOTEpisode.allEpisodes.filter {$0.season == 6}
+    //    var seasonSeven = GOTEpisode.allEpisodes.filter {$0.season == 7}
+    //    var seasonEight = GOTEpisode.allEpisodes.filter {$0.season == 8}
+    
+    var allGOT = GOTEpisode.filterAndStoreBySeason()
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view, typically from a nib.
+        tableView.dataSource = self
+        tableView.delegate = self
+        
+//        allGOT = GOTEpisode.filterAndStoreBySeason()
     }
 
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
-        // Dispose of any resources that can be recreated.
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        guard let indexPath = tableView.indexPathForSelectedRow,
+            let epDetailVC = segue.destination as? EpisodeDetailViewController else { return }
+        let episode = allGOT[indexPath.section][indexPath.row]
+        epDetailVC.delegate = episode
     }
 
 
 }
 
+extension ViewController: UITableViewDataSource, UITableViewDelegate{
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return 100
+    }
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return allGOT.count // 7, maybe 8 seasons
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return allGOT[section].count
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        switch section {
```

Comment:
This works, but instead of a switch statement you could make it more succinct and dynamic by using the sections like this 

`return "season \(section + 1)"`